pydantic_ai_slim/pydantic_ai/ui/ag_ui/_adapter.py

```diff
@@ -117,79 +127,141 @@ def state(self) -> dict[str, Any] | None:
 
         return cast('dict[str, Any]', state)
 
+    @classmethod
+    def _load_binary_part(
+        cls, part: BinaryInputContent
+    ) -> BinaryContent | ImageUrl | VideoUrl | AudioUrl | DocumentUrl:
+        """Transforms an AG-UI BinaryInputContent part into a Pydantic AI content part."""
+        if part.url:
+            try:
+                return BinaryContent.from_data_uri(part.url)
+            except ValueError:
+                media_type_constructors = {
+                    'image': ImageUrl,
+                    'video': VideoUrl,
+                    'audio': AudioUrl,
+                }
+                media_type_prefix = part.mime_type.split('/', 1)[0]
+                constructor = media_type_constructors.get(media_type_prefix, DocumentUrl)
+                return constructor(
+                    url=part.url,
+                    media_type=part.mime_type,
+                    identifier=part.id,
+                    vendor_metadata={'filename': part.filename},
```

Comment:
We can't do this this, this will be interpreted as `video_metadata` by GoogleModel